frappe/model/base_document.py

```diff
@@ -721,8 +721,13 @@ def db_insert(self, ignore_if_duplicate=False):
 		)
 
 		columns = list(d)
-		savepoint = "before_insert"
-		frappe.db.savepoint(savepoint)
+		savepoint = None
+		need_savepoint = (
+			frappe.db.db_type == "postgres" and self.meta.autoname == "hash" and not ignore_if_duplicate
+		)  # flag to indicate savepoint is needed (pg specific behavior)
```

Comment:
```suggestion
		)
```

Don't explain code in comments; it's usually redundant. 

Comments should explain "why" not "what. 


E.g. a good comment here would be:

> Postgres requires an explicit rollback to a savepoint in case of a hash collision.